File: src/co/soundwave/repository/genre/GenreDAO.kt
package co.soundwave.repository.genre

import co.soundwave.models.Genre
import co.soundwave.repository.BaseDAO

class GenreDAO : BaseDAO<GenreSQL, Genre>() {
    override val sql: GenreSQL by lazy { GenreSQL() }
    fun getInsertQuery(obj: Genre): String =
        sql.insertGenres(obj.name, obj.imgPath)
}

File: repository/genre/GenreDAO.py
from models import Genre as ge
from repository import BaseDAO as bdao
from repository.genre import GenreSQL as gesql


class GenreDAO(bdao.BaseDAO):
    def __init__(self):
        super().__init__()
        self.sql = gesql.GenreSQL()

    def get_insert_query(self, gen: ge.Genre) -> str:
        return self.sql.insert_genre(gen.name, gen.img_path)
